src/component.cpp

```diff
@@ -432,4 +432,30 @@ ComponentPtr Component::clone() const
     return c;
 }
 
+bool doRequiresImport(const ComponentPtr &thisComponent)
+{
+    if (thisComponent->isImport()) {
+        return true;
+    }
+    for (size_t c = 0; c < thisComponent->componentCount(); ++c) {
+        if (doRequiresImport(thisComponent->component(c))) {
+            return true;
+        }
+    }
+    return false;
+}
+
+bool Component::requiresImports() const
+{
+    if (isImport()) {
+        return true;
+    }
+    for (size_t c = 0; c < componentCount(); ++c) {
+        if (doRequiresImport(component(c))) {
+            return true;
+        }
+    }
+    return false;
```

Comment:
can't this all be replaced with `doRequiresImport(this)` or something like that?

Reply:
Done in commit 739d5864.